### DISCUSSION TITLE
Running MiXCR on Overlap Extension Data

### DISCUSSION THREAD

**Discussion by spracers** [milaboratory/mixcr#1394] [Q&A] [answered]
I'm hoping to run MiXCR on data collected from an overlap extension RT-PCR library. In this library, the primers were designed to be overlapping such that the TCR alpha and TCR beta transcripts become physically linked together on the same DNA construct. These constructs were then sequenced on an Illumina MiSeq.

The GitHub repository (https://github.com/milaboratory/mixcr/releases/tag/v4.1.1) states that MiXCR has implemented initial support for overlap extension based chain pairing protocols, but I haven't been able to find any documentation or guidance on how to use MiXCR on sequencing data from these overlap extension libraries.

I'd really appreciate guidance on applying MiXCR to this type of data!

**Comment by mizraelson** [comment]
Hi, can you describe your library structure in more detail? Is it always one chain (e.g. alpha) in R1 and the other (e.g. beta) is in R2? Do you have any UMIs?

**Reply by spracers** [reply]
Yes, for each sequencing construct, the TCR alpha chain is covered in read1 and the TCR beta chain is covered in read2.

Each sequencing construct (from 5' to 3') consist of the read 1 sequencing adapter, the V alpha sequence, the J alpha sequence, the C alpha sequence, the linker region (generated by the overlap in the primers), the V beta sequence, the D beta sequence, the J beta sequence, and C beta sequence, and the read 2 sequencing adapter.

The library does not have UMIs, but it does have Illumina barcodes at the 3' end of the construct.

**Reply by mizraelson** [reply]
Is the data already demultiplxed (by sample) or do you want to use the Illumina barcodes for demultiplexing?

**Reply by mizraelson** [reply, marked as answer]
Please put the attached YAML file either to ~/.mixcr/presets/ or to the directory from where you run mixcr. Then run:

```
mixcr analyze local:chain-pair-overlap \
--species mmu \
input_R{{CELLSPLIT:n}}.fastq.gz \
output
```

Put the `{{CELLSPLIT:n}}` pattern in place of a read number in your input files names.
Also, make sure to select the proper `--species`.

Let me know how it works.

Sincerely,
Mark
[chain-pair-overlap.yaml.zip](https://github.com/milaboratory/mixcr/files/13185044/chain-pair-overlap.yaml.zip)

**Reply by spracers** [reply]
I'm running MiXCR version 4.1.1, and I tried to run the above command (with the .yaml file provided), but I first got the error "No preset with name "generic-amplicon"", so I changed the present name to "generic-tcr-amplicon" in the .yaml file.

However, now I'm getting the error "No preset with name "exportClones-single-cell-no-umi-base"" . Is there another preset name I should be using, or do I need to be running a different version of MiXCR?

Also, with regards to the file names, I have two files (sample1_R1.fastq and sample1_R2.fastq). Should I be renaming the files to sample1_R1{{CELLSPLIT:n}}.fastq and sample1_R2{{CELLSPLIT:n}}.fastq?

The data is already demultiplexed by sample, so MiXCR wouldn't have to deal with the demultiplexing.

Thanks so much again for your help!

**Reply by spracers** [reply]
I installed MiXCR version 4.5.0, and the errors with the presets went away. I can now successfully run the command below:


```shell
mixcr analyze local:chain-pair-overlap \
--species hsa \
sample1_R{{CELLSPLIT:n}}.fastq \
output
``` 

using the input files sample1_R1.fastq and sample1_R2.fastq.

I'm using sequencing data from [this](https://drive.google.com/file/d/1XjZa8YyOt4KZYKQecOASi7Wk2IHKz95k/view?usp=drive_link) paper that is deposited on the sequencing read archive (SRA). The paper reports that they resolve ~500k paired TCRalpha-TCRbeta clonotypes per sample. However, when I ran MiXCR on the sequencing data, it only resolved ~15k TCRalpha chains and ~28k TCRbeta chains. Additionally, the output clones.tsv file did not link the alpha and beta chains, and instead gave separate rows for the individual chains.

I'd really appreciate your advice with troubleshooting this, as I've tried different settings but haven't been able to resolve this issue. Also, how do I tell MiXCR assemble that I want it to output clones that have *paired* TCRalpha-TCRbeta chains?

**Reply by mizraelson** [reply]
Yes, the preset is only compatible with the latest version of MiXCR. You can pair chains based on the value in the `tagValueREAD_IDX` column. Identical numbers indicate that the clones originate from the same read. If you could share the `.txt` report files generated by MiXCR, I could examine them for any issues that might be causing the loss of the expected number of clones. Please note that this feature is currently under development, and as of now, we cannot automatically filter out unpaired chains.

**Reply by spracers** [reply]
Thanks for your help! Attached are the .txt files generated by MiXCR.
[assemble.report.txt](https://github.com/milaboratory/mixcr/files/13287170/assemble.report.txt)
[refine.report.txt](https://github.com/milaboratory/mixcr/files/13287172/refine.report.txt)
[align.report.txt](https://github.com/milaboratory/mixcr/files/13287173/align.report.txt)
[qc.txt](https://github.com/milaboratory/mixcr/files/13287175/qc.txt)

[Here](https://drive.google.com/drive/folders/1JRIbjMHSfqmKIfDgWMHLmR3rPTiLiTN8?usp=sharing) are the input .fastq files in case it be helpful for you to have them.

The [paper](https://drive.google.com/file/d/1XjZa8YyOt4KZYKQecOASi7Wk2IHKz95k/view?usp=drive_link) reports ~339,000 paired TCRab clonotypes from this data, but when I ran MiXCR with the options provided above, it only returned ~154,000 paired TCRab clonotypes. I'd really appreciate your help with troubleshooting this - are there particular options with regard to the read alignment that might have to be changed?

Also when reviewing the results, I noticed that all of the chains have a readCount equal to 1 or 2, and I want to make sure I'm interpreting this correctly. Does this mean that there is only one or two sequencing reads supporting each TCR chain?

**Reply by mizraelson** [reply]
The paper does not specify the filtering criteria used to eliminate low-quality reads. There are two parameters you can adjust in MiXCR to increase the yield:

`-Massemble.cloneAssemblerParameters.badQualityThreshold=0`: This parameter will treat all alignments as high quality. By default, the MiXCR filters out reads with nucleotide qualities below 15. Setting this to 0 means no reads are discarded based on quality.

`-Massemble.cloneAssemblerParameters.minimalQuality=0` : This parameter sets the minimum quality required for a clone,defined by CDR3 sequence in this case. The default is set to 20, resulting in approximately 250,000 unique clones (amino acid CDR3 pairs). Reducing this value can increase the number of identified clones but may introduce artificial diversity due to the inclusion of lower-quality reads.

**Reply by spracers** [reply]
Thanks for your help, I'll try adjusting those settings! I had two other questions for you:

1. I ran the exact same command on MiXCR 4.4.2 and MiXCR 4.5.0 with the same input .fastq [files](https://drive.google.com/drive/folders/1c8F-35gezEWZI5DkI2sPqe6nAjWY844x?usp=drive_link), but I get different results. 

MiXCR 4.42 reports that it uses nearly 20% more of the reads in the clonotypes than MiXCR 4.5.0. It seems the reads that MiXCR 4.5.0 is not using are a result of failed mapping - why would this be the case?

MiXCR 4.4.2's output is as follows:
> Reads used in clonotypes, percent of total: 2321522 (86.14%)
> Average number of reads per clonotype: 7.39
> Reads dropped due to the lack of a clone sequence, percent of total: 20196 (0.75%)
> Reads dropped due to a too short clonal sequence, percent of total: 89 (0%)
> Reads dropped due to low quality, percent of total: 23 (0%)
> Reads dropped due to failed mapping, percent of total: 140170 (5.2%)
> Reads dropped with low quality clones, percent of total: 1198 (0.04%)
> Aligned reads processed: 2463117
> Reads used in clonotypes before clustering, percent of total: 2321637 (86.14%)
> Number of reads used as a core, percent of used: 2219383 (95.6%)
> Mapped low quality reads, percent of used: 102254 (4.4%)
> Reads clustered in PCR error correction, percent of used: 115 (0%)
> Reads pre-clustered due to the similar VJC-lists, percent of used: 0 (0%)
> Clonotypes dropped as low quality: 1127
> Clonotypes eliminated by PCR error correction: 0
> Clonotypes pre-clustered due to the similar VJC-lists: 0
> Clones dropped in post filtering: 0 (0%)
> Reads dropped in post filtering: 0.0 (0%)
> Alignments filtered by tag prefix: 0 (0%)
> TRA chains: 167315 (53.29%)
> TRA non-functional: 31487 (18.82%)
> TRB chains: 145987 (46.5%)
> TRB non-functional: 14843 (10.17%)
> IGH chains: 545 (0.17%)
> IGH non-functional: 7 (1.28%)
> IGK chains: 47 (0.01%)
> IGK non-functional: 0 (0%)
> IGL chains: 75 (0.02%)
> IGL non-functional: 10 (13.33%)

while MiXCR 4.5.0's output is as follows:

> Analysis time: 4.78m
> Final clonotype count: 313798
> Reads used in clonotypes, percent of total: 1792526 (66.51%)
> Average number of reads per clonotype: 5.71
> Reads dropped due to the lack of a clone sequence, percent of total: 20196 (0.75%)
> Reads dropped due to a too short clonal sequence, percent of total: 89 (0%)
> Reads dropped due to low quality, percent of total: 23 (0%)
> Reads dropped due to failed mapping, percent of total: 669180 (24.83%)
> Reads dropped with low quality clones, percent of total: 1299 (0.05%)
> Aligned reads processed: 2463117
> Reads used in clonotypes before clustering, percent of total: 1792526 (66.51%)
> Number of reads used as a core, percent of used: 1792038 (99.97%)
> Mapped low quality reads, percent of used: 488 (0.03%)
> Reads clustered in PCR error correction, percent of used: 0 (0%)
> Reads pre-clustered due to the similar VJC-lists, percent of used: 0 (0%)
> Clonotypes dropped as low quality: 1298
> Clonotypes eliminated by PCR error correction: 0
> Clonotypes pre-clustered due to the similar VJC-lists: 0
> Clones dropped in post filtering: 0 (0%)
> Reads dropped in post filtering: 0.0 (0%)
> Alignments filtered by tag prefix: 0 (0%)
> TRA chains: 167307 (53.32%)
> TRA non-functional: 31482 (18.82%)
> TRB chains: 145824 (46.47%)
> TRB non-functional: 14826 (10.17%)
> IGH chains: 545 (0.17%)
> IGH non-functional: 7 (1.28%)
> IGK chains: 47 (0.01%)
> IGK non-functional: 0 (0%)
> IGL chains: 75 (0.02%)
> IGL non-functional: 10 (13.33%)

2. How are nonproductive CDR3 sequences represented in the MiXCR clones.tsv output file? Do all nonproductive CDR3 sequences (with either a stop codon or a out-of-frame sequence) contain a "*" character in the `nSeqCDR3` column?

**Reply by mizraelson** [reply]
1. What commands did you use in 4.4.2 and 4.5.0? 
2. Stop codon is represented by a "*" symbol, and "_" means out-of-frame in `aaSeqCDR3`
